src/test/java/com/commercetools/sync/products/helpers/VariantReferenceResolverTest.java

```diff
@@ -26,461 +24,418 @@
+        final ProductVariantDraftBuilder productVariantDraftBuilder =
+            ProductVariantDraftBuilder.of().prices((PriceDraft) null);
 
-        final ObjectNode productReference1 = getProductReferenceWithId("existingKey");
-        final ObjectNode productReference2 = getProductReferenceWithId("randomKey");
+        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
+            .resolvePricesReferences(productVariantDraftBuilder)
+            .toCompletableFuture().join();
 
-        final AttributeDraft productReferenceAttribute =
-            getProductReferenceSetAttributeDraft("foo", productReference1, productReference2);
+        final List<PriceDraft> resolvedBuilderPrices = resolvedBuilder.getPrices();
+        assertThat(resolvedBuilderPrices).isEmpty();
+    }
 
-        final AttributeDraft resolvedAttributeDraft =
-            referenceResolver.resolveAttributeReference(productReferenceAttribute)
-                             .toCompletableFuture().join();
-        assertThat(resolvedAttributeDraft).isNotNull();
-        assertThat(resolvedAttributeDraft.getValue()).isNotNull();
+    @Test
+    void resolvePricesReferences_WithPriceReferences_ShouldResolvePrices() {
+        final CustomFieldsDraft customFieldsDraft = CustomFieldsDraft
+            .ofTypeIdAndJson("customTypeId", new HashMap<>());
 
-        final Spliterator<JsonNode> attributeReferencesIterator = resolvedAttributeDraft.getValue().spliterator();
-        assertThat(attributeReferencesIterator).isNotNull();
-        final Set<JsonNode> resolvedSet = StreamSupport.stream(attributeReferencesIterator, false)
-                                                       .collect(Collectors.toSet());
+        final PriceDraft priceDraft = PriceDraftBuilder.of(MoneyImpl.of(BigDecimal.TEN, DefaultCurrencyUnits.EUR))
+                                                       .custom(customFieldsDraft).build();
+
+        final ProductVariantDraftBuilder productVariantDraftBuilder =
+            ProductVariantDraftBuilder.of().prices(priceDraft);
 
-        final ObjectNode resolvedReference1 = getProductReferenceWithId("existingId");
-        final ObjectNode resolvedReference2 = getProductReferenceWithId("randomKey");
-        assertThat(resolvedSet).containsExactlyInAnyOrder(resolvedReference1, resolvedReference2);
+        final ProductVariantDraftBuilder resolvedBuilder = referenceResolver
+            .resolvePricesReferences(productVariantDraftBuilder)
+            .toCompletableFuture().join();
+
+        final List<PriceDraft> resolvedBuilderPrices = resolvedBuilder.getPrices();
+        assertThat(resolvedBuilderPrices).isNotEmpty();
+        final PriceDraft resolvedPriceDraft = resolvedBuilderPrices.get(0);
+        assertThat(resolvedPriceDraft).isNotNull();
+        assertThat(resolvedPriceDraft.getCustom()).isNotNull();
+        assertThat(resolvedPriceDraft.getCustom().getType().getId()).isEqualTo("typeId");
```

Comment:
maybe use constant for typeId ?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/442/commits/b936197ff501e01516f748cf9335e18f59ec1aa3